client/rest/src/main/java/org/elasticsearch/client/DeadHostState.java

```diff
@@ -47,10 +51,19 @@ private DeadHostState() {
      * that failed many consecutive times).
      */
     DeadHostState(DeadHostState previousDeadHostState) {
+        this(previousDeadHostState, TimeSupplier.DEFAULT);
```

Comment:
Yeah, I think it isn't worth having the the `TimeSupplier.DEFAULT` versions. The caller should use `TimeSupplier.DEFAULT`.